core/src/main/scala/org/http4s/parser/AcceptCharsetHeader.scala

```diff
@@ -11,35 +11,38 @@
 package org.http4s
 package parser
 
+import cats.parse.Parser1
 import cats.syntax.all._
-import org.http4s.internal.parboiled2._
 import org.http4s.CharsetRange._
-import org.http4s.QValue.QValueParser
+import org.http4s.headers.`Accept-Charset`
 
 private[parser] trait AcceptCharsetHeader {
-  def ACCEPT_CHARSET(value: String): ParseResult[headers.`Accept-Charset`] =
-    new AcceptCharsetParser(value).parse
-
-  private class AcceptCharsetParser(input: ParserInput)
-      extends Http4sHeaderParser[headers.`Accept-Charset`](input)
-      with QValueParser {
-    def entry: Rule1[headers.`Accept-Charset`] =
-      rule {
-        oneOrMore(CharsetRangeDecl).separatedBy(ListSep) ~ EOL ~> { (xs: Seq[CharsetRange]) =>
-          headers.`Accept-Charset`(xs.head, xs.tail: _*)
+  def ACCEPT_CHARSET(value: String): ParseResult[`Accept-Charset`] =
+    acceptCharsetParser.parseAll(value).leftMap { e =>
+      ParseFailure("Invalid Accept Charset header", e.toString)
+    }
+
+  private[http4s] val acceptCharsetParser: Parser1[`Accept-Charset`] = {
+    import Rfc2616BasicRules._
+    import cats.parse.Parser._
+    import org.http4s.internal.parsing.Rfc7230.token
+
+    val anyCharset = (char('*') *> QValue.parser)
+      .map(q => if (q != QValue.One) `*`.withQValue(q) else `*`)
+
+    val fromToken = (token ~ QValue.parser).mapFilter { case (s, q) =>
+      // TODO handle tokens that aren't charsets
+      Charset
```

Comment:
I think `charset` appears in multiple places, and we could put a parser on it.

Reply:
can you advise what are the other places?

Reply:
I thought there were a lot more, but maybe `Content-Type` is the last of them?

I think we can address that in a follow-up after this gets merged.

Reply:
Wow, eveything is quite interveawed :-) 
The `Content-Type` depends on the `MediaType.scala` as well. I can take this as next parser to convert. Just let me know if it makes to continue in this PR or merge first.

Reply:
Yeah, the spec is complicated, and parboiled, great as it was, didn't compose well.  I'm unsurprised we're running into tangled cords here.

I'd rather get this landed and fix that in the next.

Reply:
ok